registries/adapters/apiv2_adapter.go

```diff
@@ -0,0 +1,245 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package adapters
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/automationbroker/bundle-lib/bundle"
+	"github.com/automationbroker/bundle-lib/registries/adapters/oauth"
+	log "github.com/sirupsen/logrus"
+)
+
+const apiV2ManifestURL = "%v/v2/%v/manifests/%v"
+const apiV2CatalogURL = "%v/v2/_catalog"
+
+// OpenShiftAdapter - OpenShift Adapter
+type OpenShiftAdapter struct {
+	*apiV2Adapter
+}
+
+// PartnerRhccAdapter - Partner RHCC Adapter
+type PartnerRhccAdapter struct {
+	*apiV2Adapter
+}
+
+// apiV2Adapter - API V2 Adapter
+type apiV2Adapter struct {
+	config Configuration
+	client *oauth.Client
+}
+
+// apiV2CatalogResponse - Catalog Response
+type apiV2CatalogResponse struct {
+	Repositories []string `json:"repositories"`
+}
+
+// NewOpenShiftAdapter - creates a new OpenShift Adapter
+func NewOpenShiftAdapter(config Configuration) (*OpenShiftAdapter, error) {
+	apiV2, err := newAPIV2Adapter(config)
+	if err != nil {
+		return nil, err
+	}
+	return &OpenShiftAdapter{apiV2}, nil
+}
+
+// NewPartnerRhccAdapter - creates a new Partner RHCC Adapter
+func NewPartnerRhccAdapter(config Configuration) (*PartnerRhccAdapter, error) {
+	apiV2, err := newAPIV2Adapter(config)
+	if err != nil {
+		return nil, err
+	}
+	return &PartnerRhccAdapter{apiV2}, nil
+}
+
+// newAPIV2Adapter - creates and returns a *apiV2Adapter ready to use.
+func newAPIV2Adapter(config Configuration) (*apiV2Adapter, error) {
+	apiv2a := apiV2Adapter{
+		config: config,
+		client: oauth.NewClient(config.User, config.Pass, config.SkipVerifyTLS, config.URL),
+	}
+
+	// Authorization
+	err := apiv2a.client.Getv2()
+	if err != nil {
+		log.Errorf("Failed to GET /v2 at %s - %v", apiV2CatalogURL, err)
+		return nil, err
+	}
+
+	// set Tag to latest if empty
+	if apiv2a.config.Tag == "" {
+		apiv2a.config.Tag = "latest"
+	}
+
+	return &apiv2a, nil
+}
+
+// RegistryName - Retrieve the registry name
+func (r apiV2Adapter) RegistryName() string {
+	if r.config.URL.Host == "" {
+		return r.config.URL.Path
+	}
+	return r.config.URL.Host
+}
+
+// GetImageNames - retrieve the images
+func (r apiV2Adapter) GetImageNames() ([]string, error) {
+	log.Debugf("%s - GetImageNames", r.config.AdapterName)
+	log.Debugf("BundleSpecLabel: %s", BundleSpecLabel)
+
+	// check if we're configured for specific images
+	if r.config.Images != nil {
+		log.Debugf("Configured to use images: %v", r.config.Images)
+		return r.config.Images, nil
```

Comment:
This is fine for now but I think we wanted to also be able to include configured images AND try discovery. Thoughts?

Reply:
Should this be an issue to track this idea?

Reply:
@shawn-hurley no i don't think that'll be necessary.  as discussed with @dymurray I will do the following
- Add custom Images if list is specified 
- do discovery, and append to the image list
- If there are no images after the above two steps, return error
What do y'all think?